EngineLayer/SpectralLibrarySearch/SpectralLibrarySearchFunction.cs

```diff
@@ -67,13 +67,13 @@ public static double CalculateNormalizedSpectralAngle(List<MatchedFragmentIon> t
                 double mz = scan.MassSpectrum.XArray[closestPeakIndex];
                 double experimentalIntensity = scan.MassSpectrum.YArray[closestPeakIndex];
 
-                // is the mass error acceptable?
-                if (commonParameters.ProductMassTolerance.Within(mz.ToMass(libraryIon.Charge), libraryIon.Mz.ToMass(libraryIon.Charge)))
-                {
-                    var test = new Product(libraryIon.NeutralTheoreticalProduct.ProductType, libraryIon.NeutralTheoreticalProduct.Terminus,
+                var test = new Product(libraryIon.NeutralTheoreticalProduct.ProductType, libraryIon.NeutralTheoreticalProduct.Terminus,
```

Comment:
this will create a new Product for every library ion, even if the mass isn't within tolerance. it's more memory-efficient to leave it as the old way